code/game/objects/effects/aliens.dm

```diff
@@ -183,8 +188,9 @@
 /obj/structure/alien/weeds/Destroy()
 	var/turf/T = loc
 	loc = null
-	for (var/obj/structure/alien/weeds/W in range(1,T))
-		W.updateWeedOverlays()
+	for(var/obj/sctructure/alien/weeds/W in range(1,T))
+		if(istype(W, type_of_weed)) //so that you don't get weeds connecting to frost

```

Comment:
remove the specific nature of the comment.
just mention you don't get mixing types.